### DISCUSSION TITLE
unprintable chacters on screen

### DISCUSSION THREAD

**Discussion by vimalk78** [ghostty-org/ghostty#3770] [Issue Triage] [answered]
observed quite frequently.

```
❯ ghostty +version
0�:8�UGhostty 1.0.0-dev+0000000

Version
  - version: 1.0.0-dev+0000000
  - channel: tip
Build Config
  - Zig version: 0.13.0
  - build mode : builtin.OptimizeMode.ReleaseFast
  - app runtime: apprt.Runtime.gtk
  - font engine: font.main.Backend.fontconfig_freetype
  - renderer   : renderer.OpenGL
  - libxev     : main.Backend.io_uring
  - GTK version:
    build      : 4.16.5
    runtime    : 4.16.5
  - libadwaita : enabled
    build      : 1.6.2
    runtime    : 1.6.2
```

Discussion Number: None

## System Details Report
---

### Report details
- **Date generated:**                              2024-12-28 23:22:24

### Hardware Information:
- **Hardware Model:**                              Micro-Star International Co., Ltd. MS-7E26
- **Memory:**                                      64.0 GiB
- **Processor:**                                   AMD Ryzen™ 9 7900X × 24
- **Graphics:**                                    NVIDIA GeForce GTX 1650
- **Graphics 1:**                                  Advanced Micro Devices, Inc. [AMD®/ATI] Raphael
- **Disk Capacity:**                               2.3 TB

### Software Information:
- **Firmware Version:**                            1.E0
- **OS Name:**                                     Fedora Linux 41 (Workstation Edition)
- **OS Build:**                                    (null)
- **OS Type:**                                     64-bit
- **GNOME Version:**                               47
- **Windowing System:**                            Wayland
- **Kernel Version:**                              Linux 6.12.6-200.fc41.x86_64

**Comment by vimalk78** [comment]
installed via

```
$ sudo dnf copr enable pgdev/ghostty
$ sudo dnf install ghostty
```

**Comment by mitchellh** [comment]
Converted to discussion. Please read the rules next time

**Comment by vimalk78** [comment]
i read the rules, and was wondering why a bug needs to start as a discussion. anyways.

Is there some problem in my configuration, environment? os is this a known issue?

**Reply by mitchellh** [reply]
If you read the rules it explains exactly why.

**Reply by mitchellh** [reply]
![CleanShot 2024-12-28 at 21 46 26@2x](https://github.com/user-attachments/assets/e212f354-49ac-4ba5-871b-266992a55785)

**Reply by vimalk78** [reply]
this is a great idea.

**Comment by mitchellh** [comment]
Do you have any steps to reproduce this? I can't get it on my system and don't see any other reports about it.

Is your terminfo being recognized properly? 

What shell is this?

Do you have shell integration enabled?

**Comment by vimalk78** [comment]
> Do you have any steps to reproduce this?

there are no specific steps. just installed via copr, and started using


> Is your terminfo being recognized properly?

```
❯ echo $TERM
�lRVxterm-ghostty
```
output of `infocmp`
```
❯ infocmp
#       Reconstructed via infocmp from file: /usr/share/terminfo/x/xterm-ghostty
xterm-ghostty|ghostty|Ghostty,
        am, bce, ccc, hs, km, mc5i, mir, msgr, npc, xenl,
        colors#0x100, cols#80, it#8, lines#24, pairs#0x7fff,
        acsc=++\,\,--..00``aaffgghhiijjkkllmmnnooppqqrrssttuuvvwwxxyyzz{{||}}~~,
        bel=^G, blink=\E[5m, bold=\E[1m, cbt=\E[Z, civis=\E[?25l,
        clear=\E[H\E[2J, cnorm=\E[?12l\E[?25h, cr=\r,
        csr=\E[%i%p1%d;%p2%dr, cub=\E[%p1%dD, cub1=^H,
        cud=\E[%p1%dB, cud1=\n, cuf=\E[%p1%dC, cuf1=\E[C,
        cup=\E[%i%p1%d;%p2%dH, cuu=\E[%p1%dA, cuu1=\E[A,
        cvvis=\E[?12;25h, dch=\E[%p1%dP, dch1=\E[P, dim=\E[2m,
        dl=\E[%p1%dM, dl1=\E[M, dsl=\E]2;\007, ech=\E[%p1%dX,
        ed=\E[J, el=\E[K, el1=\E[1K, flash=\E[?5h$<100/>\E[?5l,
        fsl=^G, home=\E[H, hpa=\E[%i%p1%dG, ht=^I, hts=\EH,
        ich=\E[%p1%d@, ich1=\E[@, il=\E[%p1%dL, il1=\E[L, ind=\n,
        indn=\E[%p1%dS,
        initc=\E]4;%p1%d;rgb:%p2%{255}%*%{1000}%/%2.2X/%p3%{255}%*%{1000}%/%2.2X/%p4%{255}%*%{1000}%/%2.2X\E\\,
        invis=\E[8m, kDC=\E[3;2~, kEND=\E[1;2F, kHOM=\E[1;2H,
        kIC=\E[2;2~, kLFT=\E[1;2D, kNXT=\E[6;2~, kPRV=\E[5;2~,
        kRIT=\E[1;2C, kbs=^?, kcbt=\E[Z, kcub1=\EOD, kcud1=\EOB,
        kcuf1=\EOC, kcuu1=\EOA, kdch1=\E[3~, kend=\EOF, kent=\EOM,
        kf1=\EOP, kf10=\E[21~, kf11=\E[23~, kf12=\E[24~,
        kf13=\E[1;2P, kf14=\E[1;2Q, kf15=\E[1;2R, kf16=\E[1;2S,
        kf17=\E[15;2~, kf18=\E[17;2~, kf19=\E[18;2~, kf2=\EOQ,
        kf20=\E[19;2~, kf21=\E[20;2~, kf22=\E[21;2~,
        kf23=\E[23;2~, kf24=\E[24;2~, kf25=\E[1;5P, kf26=\E[1;5Q,
        kf27=\E[1;5R, kf28=\E[1;5S, kf29=\E[15;5~, kf3=\EOR,
        kf30=\E[17;5~, kf31=\E[18;5~, kf32=\E[19;5~,
        kf33=\E[20;5~, kf34=\E[21;5~, kf35=\E[23;5~,
        kf36=\E[24;5~, kf37=\E[1;6P, kf38=\E[1;6Q, kf39=\E[1;6R,
        kf4=\EOS, kf40=\E[1;6S, kf41=\E[15;6~, kf42=\E[17;6~,
        kf43=\E[18;6~, kf44=\E[19;6~, kf45=\E[20;6~,
        kf46=\E[21;6~, kf47=\E[23;6~, kf48=\E[24;6~,
        kf49=\E[1;3P, kf5=\E[15~, kf50=\E[1;3Q, kf51=\E[1;3R,
        kf52=\E[1;3S, kf53=\E[15;3~, kf54=\E[17;3~,
        kf55=\E[18;3~, kf56=\E[19;3~, kf57=\E[20;3~,
        kf58=\E[21;3~, kf59=\E[23;3~, kf6=\E[17~, kf60=\E[24;3~,
        kf61=\E[1;4P, kf62=\E[1;4Q, kf63=\E[1;4R, kf7=\E[18~,
        kf8=\E[19~, kf9=\E[20~, khome=\EOH, kich1=\E[2~,
        kind=\E[1;2B, kmous=\E[<, knp=\E[6~, kpp=\E[5~,
        kri=\E[1;2A, oc=\E]104\007, op=\E[39;49m, rc=\E8,
        rep=%p1%c\E[%p2%{1}%-%db, rev=\E[7m, ri=\EM,
        rin=\E[%p1%dT, ritm=\E[23m, rmacs=\E(B, rmam=\E[?7l,
        rmcup=\E[?1049l, rmir=\E[4l, rmkx=\E[?1l\E>, rmso=\E[27m,
        rmul=\E[24m, rs1=\E]\E\\\Ec, sc=\E7,
        setab=\E[%?%p1%{8}%<%t4%p1%d%e%p1%{16}%<%t10%p1%{8}%-%d%e48;5;%p1%d%;m,
        setaf=\E[%?%p1%{8}%<%t3%p1%d%e%p1%{16}%<%t9%p1%{8}%-%d%e38;5;%p1%d%;m,
        sgr=%?%p9%t\E(0%e\E(B%;\E[0%?%p6%t;1%;%?%p2%t;4%;%?%p1%p3%|%t;7%;%?%p4%t;5%;%?%p7%t;8%;m,
        sgr0=\E(B\E[m, sitm=\E[3m, smacs=\E(0, smam=\E[?7h,
        smcup=\E[?1049h, smir=\E[4h, smkx=\E[?1h\E=, smso=\E[7m,
        smul=\E[4m, tbc=\E[3g, tsl=\E]2;, u6=\E[%i%d;%dR, u7=\E[6n,
        u8=\E[?%[;0123456789]c, u9=\E[c, vpa=\E[%i%p1%dd,
```

> What shell is this?

this is bash, with starship https://starship.rs/
```
❯ echo $SHELL
�^6zWV/bin/bash
```

> Do you have shell integration enabled?

to my knowledge no. what do i check further?

**Reply by mitchellh** [reply]
Hm. All looks good. Does this reproduce with other shells?

**Reply by vimalk78** [reply]
i have not tried other shells. i can try and report.

and i found this

```
❯ ghostty +show-config --default | grep integration
��!eshell-integration = detect
shell-integration-features = cursor,no-sudo,title
```

**Reply by vimalk78** [reply]
i am not used to using shells other than bash. so can't report much.

**Comment by kshwetangi** [comment]
Okay. I’ve been following your work on the *Ghostyy* project and I’m really
impressed by the direction it's going. I believe I can contribute
meaningfully to the project, and I’d be honored to collaborate with you.

Would you consider adding me as a collaborator on the *Ghostyy* repository?
I’m excited about the potential to contribute my skills and help improve
the project in any way I can.

**Comment by pluiedev** [comment, marked as answer]
This is very likely fixed by #7808
